itests/standalone/basic/src/main/java/org/wildfly/camel/test/cdi/CDIContextCreationIntegrationTest.java

```diff
@@ -61,56 +60,57 @@ public static JavaArchive createDeployment() {
-        return ShrinkWrap.create(JavaArchive.class, name)
-            .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
-            .addPackage(InjectedContextBean.class.getPackage());
-    }
-
-    @After
-    public void tearDown() {
-        Container.Instance.set(null);
-    }
-
     @Test
     public void testCDIContextCreation() throws InterruptedException {
-        final FakeContainer container = new FakeContainer();
-        Container.Instance.set(container);
+        FakeContextTracker tracker = new FakeContextTracker();
+        try {
+            tracker.open();
+
+            Assert.assertEquals(0, tracker.getContextCount());
+            Assert.assertEquals(0, contextRegistry.getCamelContexts().size());
 
-        Assert.assertEquals(0, container.getContextCount());
-        Assert.assertEquals(0, contextRegistry.getCamelContexts().size());
+            deployer.deploy(CDI_CONTEXT_A);
+            Assert.assertEquals(1, tracker.getContextCount());
+            Assert.assertEquals(1, contextRegistry.getCamelContexts().size());
 
-        deployer.deploy(CDI_CONTEXT_A);
-        Assert.assertEquals(1, container.getContextCount());
-        Assert.assertEquals(1, contextRegistry.getCamelContexts().size());
+            deployer.deploy(CDI_CONTEXT_B);
+            Assert.assertEquals(2, tracker.getContextCount());
+            Assert.assertEquals(2, contextRegistry.getCamelContexts().size());
 
-        deployer.deploy(CDI_CONTEXT_B);
-        Assert.assertEquals(2, container.getContextCount());
-        Assert.assertEquals(2, contextRegistry.getCamelContexts().size());
+            deployer.undeploy(CDI_CONTEXT_A);
+            deployer.undeploy(CDI_CONTEXT_B);
 
-        deployer.undeploy(CDI_CONTEXT_A);
-        deployer.undeploy(CDI_CONTEXT_B);
+            Assert.assertEquals(0, contextRegistry.getCamelContexts().size());
+        } finally {
+            tracker.close();
+        }
+    }
 
-        Assert.assertEquals(0, contextRegistry.getCamelContexts().size());
+    @Deployment(name = CDI_CONTEXT_A, managed = false, testable = false)
+    public static JavaArchive createCdiTestJar() {
+        JavaArchive archive = ShrinkWrap.create(JavaArchive.class, CDI_CONTEXT_A);
+        archive.addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+        archive.addPackage(InjectedContextBean.class.getPackage());
+        return archive;
+    }
+
+    @Deployment(name = CDI_CONTEXT_B, managed = false , testable = false)
+    public static JavaArchive createOtherCdiTestJar() {
+        JavaArchive archive = ShrinkWrap.create(JavaArchive.class, CDI_CONTEXT_B);
+        archive.addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+        archive.addPackage(InjectedContextBean.class.getPackage());
+        return archive;
     }
 
-    private static final class FakeContainer implements Container {
+    private class FakeContextTracker extends CamelContextTracker {
```

Comment:
Why would we not want to verify the interaction with the camel 'container' on CDI context creation?, given it was one of the root causes of the problems documented in https://github.com/wildfly-extras/wildfly-camel/issues/1107